### DISCUSSION TITLE
App state management

### DISCUSSION THREAD

**Discussion by claritypsychology** [ariakit/ariakit#811] [Questions] [answered]
How do I control the state of one component from another when they are in separate modules. 

For example i have a dialog component included in my page.  Also included in the page is a list. That list is made up of button components.  I want each of those button components in the list to be able to open the dialog in the page.

To manage their state within the application i am using context.  However if i use the context to manage the visibility of the modal I lose the esc/click outside to close and focus back on the disclosure functionality.

All help appreciated.

**Comment by diegohaz** [comment, marked as answer]
You can use this approach: https://reakit.io/docs/managing-state/#shared-state
